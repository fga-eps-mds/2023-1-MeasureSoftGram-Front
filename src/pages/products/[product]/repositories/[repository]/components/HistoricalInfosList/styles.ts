import styled from 'styled-components';

export const HistoricalInfosBackground = styled.div`
  padding: 15px 0px;

  background-color: white;
`;