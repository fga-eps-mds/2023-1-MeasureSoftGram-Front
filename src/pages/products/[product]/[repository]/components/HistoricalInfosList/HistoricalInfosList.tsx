import React from 'react';
import { Container } from '@mui/material';
import HistoricalLatestInfos from './HistoricalLatestInfos';
import * as Styles from './styles';

interface OptionCheckedProps {
  [key: string]: boolean;
}

interface Prop {
  checkedOptions: OptionCheckedProps;
  currentRepository: any;
}

const HistoricalInfosList = ({ checkedOptions, currentRepository }: Prop) => (
  <Styles.HistoricalInfosBackground>
    <Container>
      <HistoricalLatestInfos checkedOptions={checkedOptions} currentRepository={currentRepository} />
    </Container>
  </Styles.HistoricalInfosBackground>
);

export default HistoricalInfosList;
