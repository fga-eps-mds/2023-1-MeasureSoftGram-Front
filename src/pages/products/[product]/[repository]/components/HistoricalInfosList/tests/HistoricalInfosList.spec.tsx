import React from 'react';
import { render } from '@testing-library/react';
import HistoricalInfosList from '../HistoricalInfosList';

jest.mock('@contexts/ProductProvider', () => ({
  useProductContext: () => ({
    currentProduct: {
      name: `aoba`
    }
  })
}));

describe('HistoricalInfosList', () => {
  it('renders correctly', () => {
    const checkedOptions = {
      option1: true,
      option2: false
    };
    const { container } = render(<HistoricalInfosList checkedOptions={checkedOptions} />);
    expect(container).toMatchSnapshot();
  });
});
