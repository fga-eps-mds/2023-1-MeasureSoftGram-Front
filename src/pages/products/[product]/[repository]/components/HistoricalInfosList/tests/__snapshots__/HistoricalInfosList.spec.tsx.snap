// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoricalInfosList renders correctly 1`] = `
<div>
  <div
    class="styles__HistoricalInfosBackground-sc-17ev0um-0 hYTmFv"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-kzcie9"
      >
        <div
          class="styles__FilterBackground-sc-n9f0ab-0 kGdXmE"
        >
          <div
            class="MuiBox-root css-1k25d3d"
          >
            <div
              class="MuiBox-root css-l57rop"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                data-testid="shimmer"
                style="width: 200px; height: 32px;"
              />
              <div
                class="MuiBox-root css-vqe3zw"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                  data-testid="shimmer"
                  style="width: 180px; height: 25px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                  data-testid="shimmer"
                  style="width: 180px; height: 25px;"
                />
              </div>
            </div>
            <div
              class="MuiBox-root css-l57rop"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                data-testid="shimmer"
                style="width: 200px; height: 32px;"
              />
              <div
                class="MuiBox-root css-vqe3zw"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                  data-testid="shimmer"
                  style="width: 180px; height: 25px;"
                />
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                  data-testid="shimmer"
                  style="width: 180px; height: 25px;"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div
            class="MuiBox-root css-1ypheah"
          >
            <div
              class="MuiBox-root css-0"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                data-testid="shimmer"
                style="width: 400px; height: 50px;"
              />
              <span
                class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                data-testid="shimmer"
                style="width: 200px; height: 16px;"
              />
            </div>
          </div>
          <div
            class="MuiBox-root css-0"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-58ftfq-MuiSkeleton-root"
              data-testid="shimmer"
              style="height: 280px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
